controllers/clustercache/cluster_accessor.go

```diff
@@ -495,3 +497,80 @@ func (ca *clusterAccessor) unlock(ctx context.Context) {
 	ca.lockedStateLock.Unlock()
 	log.V(10).Info("Removed lock for ClusterAccessor")
 }
+
+// ScopedWatcher is a scoped-down interface from Controller that only has the Watch func.
+type ScopedWatcher[request comparable] interface {
+	Watch(src source.TypedSource[request]) error
+}
+
+// WatchInput specifies the parameters used to establish a new watch for a workload cluster.
+// A source.TypedKind source (configured with Kind, TypedEventHandler and Predicates) will be added to the Watcher.
+// To watch for events, the source.TypedKind will create an informer on the Cache that we have created and cached
+// for the given Cluster.
+type WatchInput = TypedWatchInput[client.Object, ctrl.Request]
+
+// TypedWatchInput specifies the parameters used to establish a new watch for a workload cluster.
+// A source.TypedKind source (configured with Kind, TypedEventHandler and Predicates) will be added to the Watcher.
+// To watch for events, the source.TypedKind will create an informer on the Cache that we have created and cached
+// for the given Cluster.
+type TypedWatchInput[object client.Object, request comparable] struct {
+	// Name represents a unique Watch request for the specified Cluster.
+	// The name is used to track that a specific watch is only added once to a cache.
+	// After a connection (and thus also the cache) has been re-created, watches have to be added
+	// again by calling the Watch method again.
+	Name string
+
+	// Watcher is the watcher (controller) whose Reconcile() function will be called for events.
+	Watcher ScopedWatcher[request]
+
+	// Kind is the type of resource to watch.
+	Kind object
+
+	// EventHandler contains the event handlers to invoke for resource events.
+	EventHandler handler.TypedEventHandler[object, request]
+
+	// Predicates is used to filter resource events.
+	Predicates []predicate.TypedPredicate[object]
+}
+
+// NewWatcher creates a Watcher on the workload cluster.
+// A source.TypedKind source (configured with Kind, TypedEventHandler and Predicates) will be added to the ScopedWatcher.
+// To watch for events, the source.TypedKind will create an informer on the Cache that we have created and cached
+// for the given Cluster.
+func NewWatcher(input WatchInput) Watcher {
```

Comment:
Do we still need this? Could we only have NewTypedWatcher? (and call it NewWatcher)

As far as I can tell with the current version of the PR it's already possible to do the following:
```go
	return r.ClusterCache.Watch(ctx, util.ObjectKey(cluster), clustercache.NewTypedWatcher(clustercache.WatchInput{
		Name:         "machine-watchNodes",
		Watcher:      r.controller,
		Kind:         &corev1.Node{},
		EventHandler: handler.EnqueueRequestsFromMapFunc(r.nodeToMachine),
	}))
```